engine/src/main/java/org/apache/apex/engine/YarnAppLauncherImpl.java

```diff
@@ -142,8 +142,7 @@ public String getApplicationId()
     @Override
     public boolean isFinished()
     {
-      YarnClient yarnClient = YarnClient.createYarnClient();
-      try {
+      try (YarnClient yarnClient = StramClientUtils.createYarnClient(conf)) {
```

Comment:
Will it be better to cache yarnClient on the YarnAppHandleImpl instead of creating it every time? In ApexCli the yarnClient is cached.